### DISCUSSION TITLE
Are there possibilities to get raster data

### DISCUSSION THREAD

**Discussion by wiesehahn** [open-meteo/open-meteo#309] [General]
To my basic understanding so far its possible to query data for single locations. 
But it is not possible to query data for spatial extents, or is it?

What I am looking for is some way to retrieve the data for a certain in a rasterized format e.g. as Cloud Optimized GeoTiff via HTTP Get request.

**Comment by patrick-zippenfenig** [comment]
This is correct. Open-Meteo is based on raster GRIB2 weather forecasts and is designed to access long time-series data for single locations. In the future, I will add API endpoints to get data for a list of locations, but not in raster formats.

I would not exclude to build at some point raster API and maps, but this will be a larger project. It will also double the size of the database behind Open-Meteo, as data needs to be stored optimised for spatial access. Might be possible with more collaborators and funding or in some kind of EU funded project.

As a short term solution you could use the ICON GRIB files directly from: https://opendata.dwd.de/weather/nwp/

Which kind of weather variables and area do you need? What is the use case?

**Reply by wiesehahn** [reply]
Thanks, 
I am just evaluating some possibilities regarding risk modelling of forest stands in Germany. So I am mainly interested in long term climate predictions, but weather forecasts of e.g. precipitation and wind would be also interesting.

**Reply by patrick-zippenfenig** [reply]
Data in the [Climate API](https://open-meteo.com/en/docs/climate-api) follows a simple 0.1° grid. For your study areas, you can just loop over the coordinates. Please set "&cell_selection=nearest" and "&elevation=NaN" in the URL to disable any terrain correction.

This should be relatively fast. Before you start, just give me heads up, so I do not accidentally block you IP address ;-)

Alternatively you could try to download IPCC climate models directly from [ESGF](https://esgf-node.llnl.gov/projects/cmip6/) in NetCDF format, but data won't be BIAS corrected with any ground truth data.

**Comment by wiesehahn** [comment]
If I get this correct https://github.com/open-meteo/open-data/pull/28 adds support for this, or?

**Reply by patrick-zippenfenig** [reply]
Moving towards it, but this is just the first step to generate required data. 

Ideally, I do not want to build APIs to return raster data, but instead users access the underlaying OM-files directly from an HTTP / S3 endpoint. By chunking data into small pieces, users can get the required data without downloading the entire file (cloud-native). This approach scales much better, because no CPU-expensive application servers are required on Open-Meteo side.

Next steps are:
1. Finalise [Python](https://github.com/open-meteo/python-omfiles) and [TypeScript](https://github.com/open-meteo/typescript-omfiles) libraries
2. Figure out how to properly work with grids and projections on client side
3. Write dummy code / web UI to make this data accessible to anyone. Ideally with an automatic code generator, suitable for simple climate sciences

There is no ETA. It is not the kind to work anyone is willing to pay money for. However, I think it might already be in a testable state in just a couple of weeks (@terraputix ;-))
